src/base/util_func.c

```diff
@@ -808,6 +809,35 @@ util_bsearch (const void *key, const void *base, int n_elems, unsigned int sizeo
   return mid;
 }
 
+/* util_gettime_msec - returns current time in milliseconds
+ *
+ * NOTE: currently using gettimeofday; for portability, must be implemented using
+ * clock_gettime (which, as of now, does not have an implementation on Win32)
+ */
+int64_t
+util_gettime_msec ()
+{
+  struct timeval now;
+  gettimeofday (&now, NULL);
+
+  const int64_t msec_from_sec = now.tv_sec * 1000LL;
+  const int64_t msec_from_usec = now.tv_usec / 1000LL;
+  // *INDENT-OFF*
+  // overflow check
+  assert (std::numeric_limits<int64_t>::max () - msec_from_usec >= msec_from_sec);
+  // *INDENT-ON*
+  const int64_t msec = msec_from_sec + msec_from_usec;
+  return msec;
```

Comment:
Suggestion: use <chrono> 
```suggestion
  // *INDENT-OFF*
  auto now_timept = std::chrono::system_clock::now ();
  return std::chrono::duration_cast<std::chrono::milliseconds> (now_timept .time_since_epoch ()).count ();
  // *INDENT-ON*
```

Reply:
Done